### ISSUE TITLE
[Bug] transfer_ecu_data ignores UDS negative responses to TransferData 

### ISSUE THREAD

**Issue by alexmohr** [eclipse-opensovd/classic-diagnostic-adapter#545]
## Description

A terminal NRC — e.g. 0x24 (requestSequenceError), 0x31 (requestOutOfRange), 0x72 (generalProgrammingFailure) — does not abort the flash transfer. The transfer continues sending subsequent blocks and finishes with `status: "finished"` and `acknowledgedBytes == length`, so a flash that the ECU rejected in full is reported to the SOVD client as successful.

Root cause: negative responses are not an `Err` on the send path. [`convert_from_uds`](https://github.com/eclipse-opensovd/classic-diagnostic-adapter/blob/44147e7169dcfd50a16e7958a7b7126e544449a3/cda-core/src/diag_kernel/payload_decode.rs#L230-L245) returns `Ok(response)` with `response_type == DiagServiceResponseType::Negative`. The [transfer loop](https://github.com/eclipse-opensovd/classic-diagnostic-adapter/blob/44147e7169dcfd50a16e7958a7b7126e544449a3/cda-comm-uds/src/data_transfer.rs#L136-L154) only inspects the `Err` arm and never checks `response_type()`:

```rust
let result = self
    .send(ecu_name, request.clone(), &skip_security_plugin_check, Some(uds_payload), true)
    .await;
if let Err(e) = result {
    set_transfer_aborted(/* ... */).await;
    break;
}
// falls through here on a negative response and advances the transfer
```

The block that follows unconditionally advances `next_block_sequence_counter` and adds `bytes_to_read` to `acknowledged_bytes`.

Every other UDS path in the CDA handles this correctly — the synchronous endpoints route negative responses through [`api_error_from_diag_response`](https://github.com/eclipse-opensovd/classic-diagnostic-adapter/blob/44147e7169dcfd50a16e7958a7b7126e544449a3/cda-sovd/src/sovd/error.rs#L433-L452) and return HTTP 502. Only the asynchronous flash transfer misses it.

## Steps to Reproduce

Variant A — no simulator changes needed, provokes NRC 0x31 from a conforming ECU:

1. `PUT /components/{ecu}/x-sovd2uds-download/requestdownload` and note `MaxNumberOfBlockLength` in the response.
2. `POST /components/{ecu}/x-sovd2uds-download/flashtransfer` with a `blocksize` **larger** than `MaxNumberOfBlockLength`, e.g. `{"blocksequencecounter": 1, "blocksize": 4096, "offset": 0, "length": <file_size>, "id": "<file_id>"}`, when the ECU reported a smaller maximum.
3. Poll `GET /components/{ecu}/x-sovd2uds-download/flashtransfer/{id}` until `status` is no longer `running`.

Variant B — deterministic, isolates the NRC handling:

1. Configure the ECU simulator to answer `0x36` with `0x7F 0x36 0x72` (generalProgrammingFailure).
2. Run the normal sequence: `requestdownload` → `POST flashtransfer` → poll status.

## Expected Behavior

The first negative response to a `TransferData` request aborts the transfer:

- `status` becomes `"aborted"`,
- `error` carries the NRC and a description of it,
- `acknowledgedBytes` reflects only blocks the ECU actually acknowledged positively,
- no further `0x36` blocks are transmitted.

Per ISO 14229-1, a client shall not continue a block sequence after the server rejects a block, and NRC 0x24 in particular means the sequence counter is already out of step, so continuing is guaranteed to be wrong.

## Actual Behavior

- The negative response is discarded; not even a warning is logged.
- `next_block_sequence_counter` and `acknowledgedBytes` advance as if the block had been accepted.
- The remaining blocks are transmitted and rejected in turn.
- The transfer ends with `status: "finished"`, `acknowledgedBytes == length`, and no `error` field.
- A client following the documented flow has no way to tell a successful flash from a fully rejected one.

## Screenshots or Logs (if applicable)

None captured — this was found by code inspection while investigating #540, not by a live run, so the behaviour above is derived from the code path rather than from a reproduction. No existing test covers it: [`test_flash_download_transfer_sequence`](https://github.com/eclipse-opensovd/classic-diagnostic-adapter/blob/44147e7169dcfd50a16e7958a7b7126e544449a3/integration-tests/tests/sovd/flash_download.rs#L352) only asserts the transfer is *not* aborted, which this bug trivially satisfies.

## Environment

- OS: platform-independent (pure protocol-handling logic, no OS-specific code involved)
- Version: `main` @ [`44147e7`](https://github.com/eclipse-opensovd/classic-diagnostic-adapter/tree/44147e7169dcfd50a16e7958a7b7126e544449a3)

## Additional Context

Surfaced while answering #540 (Scenario 2).

Suggested fix, in two parts:

1. In [`transfer_ecu_data`](https://github.com/eclipse-opensovd/classic-diagnostic-adapter/blob/44147e7169dcfd50a16e7958a7b7126e544449a3/cda-comm-uds/src/data_transfer.rs#L136-L154), check `response.response_type()` on the `Ok` arm and call `set_transfer_aborted` on `Negative`, reusing `MappedNRC` so the abort reason names the NRC rather than a free-text string.
2. Extend [`DataTransferError`](https://github.com/eclipse-opensovd/classic-diagnostic-adapter/blob/44147e7169dcfd50a16e7958a7b7126e544449a3/cda-sovd-interfaces/src/components/ecu.rs#L255-L257) beyond its single `text` field so clients can read the NRC programmatically, mirroring what `nrc_to_api_error_response` already exposes on the synchronous endpoints (`parameters.NRC`, `parameters.SID`, `error_source: "ECU"`).

While fixing (1), the abort reason string `format!("Failed to read data: {e:?}")` at [data_transfer.rs#L149](https://github.com/eclipse-opensovd/classic-diagnostic-adapter/blob/44147e7169dcfd50a16e7958a7b7126e544449a3/cda-comm-uds/src/data_transfer.rs#L149) is worth correcting too — it is used for UDS send failures as well as file-read failures, so a timeout currently surfaces to the client as "Failed to read data: Timeout".

A regression test belongs alongside `test_flash_download_transfer_sequence`: simulator answers 0x36 with an NRC, assert `status == "aborted"` and that `acknowledgedBytes < length`.

> [!IMPORTANT]
> Link the pull request that addresses this bug once created.
